nav2_controller/include/nav2_controller/plugins/simple_goal_checker.hpp

```diff
@@ -58,7 +58,7 @@ class SimpleGoalChecker : public nav2_core::GoalChecker
   SimpleGoalChecker();
   // Standard GoalChecker Interface
   void initialize(
-    const rclcpp_lifecycle::LifecycleNode::SharedPtr & nh,
+    const rclcpp_lifecycle::LifecycleNode::WeakPtr & parent,
```

Comment:
For all instances: I don't think its generally best practice to pass smart pointers by reference. Weak ptrs might be different, but I don't see any documentation about that. Why not have the weakptr be by value?

Reply:
From this comment: https://github.com/ros-planning/navigation2/pull/1900#discussion_r470853609 what you say makes sense to me, but the conventions around smart pointers are the opposite. We should discuss this and then resolve one way or another.